### ISSUE TITLE
dotnet run breaks argument passing to app

### ISSUE THREAD

**Issue by bojanrajkovic** [dotnet/cli#8302]
## Steps to reproduce

1. Download the attached sample.
2. Run `dotnet run -- -tfoo -`

Sample: [dotnet-run-argument-issues-repro.zip](https://github.com/dotnet/cli/files/1592588/dotnet-run-argument-issues-repro.zip)

## Expected  behavior

The following output:

```
Arguments: [-tfoo, -]
Remaining: [-]
Test argument value: foo
```

This correct behavior can be obtained by running `dotnet bin/Debug/netcoreapp2.0/dotnet-run-argument-issues-repro.dll -tfoo -`.

## Actual behavior

```
Arguments: [-tfoo]
Remaining: []
Test argument value: foo
```

The `-` is being eaten by `dotnet run`, despite the fact it's been told to stop argument parsing (`--`). When the app is shipped, this isn't a huge issue, but it's massively annoying during development.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.2)

Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  17.10
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /usr/share/dotnet/sdk/2.0.2/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

**Comment by peterhuene** [comment]
Hi @bojanrajkovic, thank you for reporting this issue!

I believe the fix for #7863 (PR dotnet/CliCommandLineParser#80) also addressed this issue.  The command line options parser now completely passes everything after the double dash as-is without attempting to parse it.

I'm going to close this issue as fixed in an upcoming release.  Please let me know if you have any questions or concerns.  Thanks!

**Comment by bojanrajkovic** [comment]
That did the trick, thanks! Definitely fixed on 2.1.300-preview2-008173/2.1.0-preview2-26131-06 here.
